File: Linguagem Java/Questão9.java
//9. Interfaces/Protocolos Em Java e Golang, defina uma interface para imprimível que tenha um método imprimir. Implemente essa interface em classes como Relatório e Contrato. Em Python, use a abordagem de protocolo ou classes abstratas.

interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    @Override
    public void imprimir() {
        System.out.println("Realizando a Impressão.");
    }
}

class Contrato implements Imprimivel {
    @Override
    public void imprimir() {
        System.out.println("Realizando a Impressão.");
    }
}

public class Main {
    public static void main(String[] args) {
        Imprimivel relatorio = new Relatorio();
        Imprimivel contrato = new Contrato();

        relatorio.imprimir();
        contrato.imprimir();
    }
}

File: Linguagem Python/Questão9.py
#9. Interfaces/Protocolos Em Java e Golang, defina uma interface para imprimível que tenha um método imprimir. Implemente essa interface em classes como Relatório e Contrato. Em Python, use a abordagem de protocolo ou classes abstratas.

from abc import ABC, abstractmethod

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def imprimir(self):
        print("Realizando Impressão.")

class Contrato(Imprimivel):
    def imprimir(self):
        print("Realizando Impressão.")

def main():
    relatorio = Relatorio()
    contrato = Contrato()
    relatorio.imprimir()
    contrato.imprimir()

if __name__ == "__main__":
    main()
